libdrizzle/field.cc

```diff
@@ -172,6 +172,12 @@ drizzle_field_t drizzle_field_buffer(drizzle_result_st *result, size_t *total,
     current_field= result->field_current-1;
   }
 
+  if (result->field_buffer_sizes[current_field] == 0)
+  {
+    result->field_buffer[current_field]= (drizzle_field_t) malloc(64*1024);
+    result->field_buffer_sizes[current_field]= 64*1024;
```

Comment:
Allocating 64KB will definitely solve the realloc problem but it will use so much ram it will probably cause the machine to go into swap. 
I'm a bit disturbed by this. If the fields was only set once, and then reallocated every time it exceeded the size, it would only reallocate a few times at startup, and then it would never happen again, so I doubt you would notice.
This sounds as though it is being allocated very often.